openedx/core/djangoapps/profile_images/images.py

```diff
@@ -127,51 +120,102 @@ def validate_uploaded_image(uploaded_file):
     string_io = StringIO()
-    scaled.save(string_io, format='JPEG')
+    if exif is None:
+        image_obj.save(string_io, format='JPEG')
+    else:
+        image_obj.save(string_io, format='JPEG', exif=exif)
     image_file = ContentFile(string_io.getvalue())
     return image_file
 
 
-def create_profile_images(image_file, profile_image_names):
+def _get_corrected_exif(image, original):
     """
-    Generates a set of image files based on image_file and
-    stores them according to the sizes and filenames specified
-    in `profile_image_names`.
+    If the original image contains exif data, use that data to
+    preserve image orientation in the new image.
     """
-    image_obj = Image.open(image_file)
+    if 'exif' in original.info:
+        image_exif = image.info.get('exif', piexif.dump({}))
+        original_exif = original.info['exif']
+        image_exif = _update_exif_orientation(image_exif, _get_exif_orientation(original_exif))
+        return image_exif
 
-    # first center-crop the image if needed (but no scaling yet).
-    width, height = image_obj.size
-    if width != height:
-        side = width if width < height else height
-        image_obj = image_obj.crop(((width - side) / 2, (height - side) / 2, (width + side) / 2, (height + side) / 2))
 
-    storage = get_profile_image_storage()
-    for size, name in profile_image_names.items():
-        scaled_image_file = _get_scaled_image_file(image_obj, size)
-        # Store the file.
-        try:
-            storage.save(name, scaled_image_file)
-        finally:
-            scaled_image_file.close()
+def _update_exif_orientation(exif, orientation):
+    """
+    Given an exif value and an integer value 1-8, reflecting a valid value for
+    the exif orientation, return a new exif with the orientation set.
+    """
+    exif_dict = piexif.load(exif)
+    exif_dict['0th'][piexif.ImageIFD.Orientation] = orientation
+    return piexif.dump(exif_dict)
 
 
-def remove_profile_images(profile_image_names):
+def _get_exif_orientation(exif):
+    """Return the orientation value for the given Image object"""
+    exif_dict = piexif.load(exif)
+    return exif_dict['0th'].get(piexif.ImageIFD.Orientation)
+
+
+def _get_valid_file_types():
     """
-    Physically remove the image files specified in `profile_image_names`
+    Return comma separated string of valid file types.
     """
-    storage = get_profile_image_storage()
-    for name in profile_image_names.values():
-        storage.delete(name)
+    return ', '.join([', '.join(IMAGE_TYPES[ft].extensions) for ft in IMAGE_TYPES.keys()])
+
+
+def _user_friendly_size(size):
+    """
+    Convert size in bytes to user friendly size.
+
+    Arguments:
+        size (int): size in bytes
+
+    Returns:
+        user friendly size
+    """
+    units = [_('bytes'), _('KB'), _('MB')]
+    i = 0
+    while size >= 1024:
+        size /= 1024
+        i += 1
+    return u'{} {}'.format(size, units[i])
```

Comment:
Seems unlikely, but certainly possible.  I'll put in a break when it reaches MB.